client/src/lib/result.js

```diff
@@ -40,3 +47,83 @@ export function paddedAttemptResults(result, numberOfAttempts) {
   const attemptResults = result.attempts.map((attempt) => attempt.result);
   return padSkipped(attemptResults, numberOfAttempts);
 }
+
+// Events sorted by best don't need forecast view.
+// Fewest moves is currently unsupported
+// Only final rounds are supported
+export function forecastViewEnabled(round) {
+  return round.format.sortBy != "best" &&
+    round.competitionEvent.event.id != "333fm" &&
+    round.advancementCondition === null;
+}
+
+export function resultProjectedAverage(result, format) {
+  if (isComplete(result.average)) {
+    return result.average;
+  }
+  return projectedAverage(result.attempts.map((attempt) => attempt.result), format);
+}
+
+/**
+ * return the results object with additional properties:
+ * projectedAverage: projected final average based on current results
+ * forFirst: time needed to overtake first place
+ * forThird: time needed to overtake third place
+ */
+export function resultsForView(results, format, forecastView) {
+  console.log(results);
+  if (results.length == 0 || !forecastView) return results;
+  var resultsForView = results.map((result) => {
+    return {
+      ...result,
+      projectedAverage: resultProjectedAverage(result, format),
+    };
+  });
+
+  // Sort based on projection with tiebreakers on single
+  resultsForView = orderBy(resultsForView, [
+    (result) => toMonotonic(result.projectedAverage),
+    (result) => toMonotonic(result.best),
+  ]);
+
+  if (isSkipped(resultsForView[0].projectedAverage)) {
+    // First place has no results. Do nothing.
```

Comment:
Looking on the comment, we can make this more explicit:

```suggestion
  if (resultsForView[0].attempts.length === 0)) {
```